src/textAngularSetup.js

```diff
@@ -565,16 +565,30 @@ angular.module('textAngularSetup', [])
 			var urlPrompt;
 			urlPrompt = $window.prompt(taTranslations.insertVideo.dialogPrompt, 'https://');
 			if (urlPrompt && urlPrompt !== '' && urlPrompt !== 'https://') {
+				var videoId = null;
 				// get the video ID
-				var ids = urlPrompt.match(/(\?|&)v=[^&]*/);
```

Comment:
I’m certainly open to this idea. I’ll admit that I’m not a regex expert so I wouldn’t be able to come up with something like that on my own. As it stands, trying to incorporate that regex is throwing the linter into a frenzy. 

 

Attempted to change it to:

 

‘’’

var regex = /(?:youtube(?:-nocookie)?.com\/(?:[^/]+\/.+\/|(?:v|e(?:mbed)?)\/|.*[?&]v=)|youtu.be\/)([^"&?/ ]{11})/i;

‘’’

 

However, this seemed to throw the code coverage tool into a tizzy saying that it could not actually get into the if-block on the matches. I’m not entirely sure what value it’s trying to throw in there, but it appears to not be able to evaluate that the path is ever crossed.

 

I’d also like to provide unit tests for all of your example urls, but the place where I am invoking the test does not appear to have a convenient way for me to do so.

 

From: Nick [mailto:notifications@github.com] 
Sent: Thursday, July 09, 2015 11:42 AM
To: fraywing/textAngular
Cc: jeremy-compeat
Subject: Re: [textAngular] (feat): change to insertVideo to handle youtu.be links (#710)

 

In src/textAngularSetup.js <https://github.com/fraywing/textAngular/pull/710#discussion_r34275465> :

>                              // get the video ID
> -                            var ids = urlPrompt.match(/(\?|&)v=[^&]*/);

Instead of having to create if cases for each version of youtbe url variatio you could do somthing like

var videoId = null;
var re = new RegExp('(?:youtube(?:-nocookie)?\.com/(?:[^/]+/.+/|(?:v|e(?:mbed)?)/|.*[?&]v=)|youtu\.be/)([^"&?/ ]{11})','i');
matches = urlPrompt.match(re):
if (match) {
  videoId = match[0]
}

This should handle:

http://www.youtube.com/watch?v=0zM3nApSvMg <http://www.youtube.com/watch?v=0zM3nApSvMg&feature=feedrec_grec_index> &feature=feedrec_grec_index
http://www.youtube.com/user/IngridMichaelsonVEVO#p/a/u/1/QdK8U-VIH_o
http://www.youtube.com/v/0zM3nApSvMg?fs=1 <http://www.youtube.com/v/0zM3nApSvMg?fs=1&amp;hl=en_US&amp;rel=0> &amp;hl=en_US&amp;rel=0
http://www.youtube.com/watch?v=0zM3nApSvMg#t=0m10s
http://www.youtube.com/embed/0zM3nApSvMg?rel=0
http://www.youtube.com/watch?v=0zM3nApSvMg
http://youtu.be/0zM3nApSvMg

—
Reply to this email directly or view it on GitHub <https://github.com/fraywing/textAngular/pull/710/files#r34275465> .  <https://github.com/notifications/beacon/AHiXWZKSHBFeU6wHzLI_baTwCVKHMZiOks5obpvrgaJpZM4Er8qt.gif>